packages/web/src/lib/web-mode.ts

```diff
@@ -184,62 +185,52 @@ export class WebMode {
 
 	/**
 	 * Appends signature to a multisig transaction using algosigner
-	 * @param txns array of multisignature transaction with at least one signature
+	 * @param txn Multisignature Encoded Transaction  
 	 * @param signers a subset of addresses to sign the transaction
-	 * return an object containing a blob attribute encoded in base64
+	 * return an object containing a blob key encoded in base64
 	 */
 	async appendSignMultisigTransaction(
-		txns: WalletTransaction[],
-		signers: string[]
+		txn: EncodedSignedTransaction,
+		signers?: string[]
 	): Promise<JsonPayload> {
 		try {
-			const result: JsonPayload = {};
-			for (let i = 0; i < txns.length; ++i) {
-				const txn = txns[i];
-				const partialTxn = algosdk.decodeObj(
-					this.algoSigner.encoding.base64ToMsgpack(txn.txn)
-				) as EncodedSignedTransaction;
-				if (partialTxn.txn === undefined || partialTxn.msig === undefined) {
-					throw new Error(
-						"Input transaction must be multisigature transaction signed with at least 1 signature"
-					);
-				}
-				const txnToBeSign = algosdk.Transaction.from_obj_for_encoding(partialTxn.txn);
-				const txnToBeSign_Uint8Array = algosdk.encodeObj(txnToBeSign.get_obj_for_encoding());
-				const txnToBeSign_Base64 =
-					this.algoSigner.encoding.msgpackToBase64(txnToBeSign_Uint8Array);
-
-				const mparams = partialTxn.msig as algosdk.EncodedMultisig;
-				const addrs = mparams.subsig.map((signData) => {
-					return algosdk.encodeAddress(signData.pk);
-				});
+			if (!txn.msig) {
+				throw new Error("Current transaction is not a Multisig Transaction.")
+			}
+			const encodedTxn = this.algoSigner.encoding.msgpackToBase64(encodeObj(txn.txn))
+			const mparams = txn.msig as algosdk.EncodedMultisig;
+			const version = mparams.v;
+			const threshold = mparams.thr;
+			const addr = mparams.subsig.map((signData) => {
+				return algosdk.encodeAddress(signData.pk)
+			});
+
+			const multisigParams = {
+				version: version,
+				threshold: threshold,
+				addrs: addr,
```

Comment:
typo error. `address`

Reply:
.? Algosigner expects the multisig params with these keys only
https://github.com/scale-it/algo-builder/blob/5581a5663dc304650fd83e0bfbadc9ad743578c9/packages/web/src/algo-signer-types.ts#L31

Reply:
ok